llvm/lib/Transforms/Scalar/LICM.cpp

```diff
@@ -2735,6 +2737,12 @@ static bool hoistMulAddAssociation(Instruction &I, Loop &L,
   if (Changes.empty())
     return false;
 
+  // Drop the poison flags for any adds we looked through.
+  if (I.getType()->isIntOrIntVectorTy()) {
+    for (auto *Add : Adds)
+      Add->dropPoisonGeneratingFlags();
```

Comment:
I think the flag drop on the adds does not have test coverage?

Reply:
Aren't they covered by the NOT_CONSTRAINED-NEXT lines? On the test case where I added `nsw nuw` to adds?